pkg/splunk/enterprise/indexercluster.go

```diff
@@ -97,24 +119,12 @@ func ApplyIndexerCluster(client splcommon.ControllerClient, cr *enterprisev1.Ind
 		return result, err
 	}
 
-	namespacedName := types.NamespacedName{
-		Namespace: cr.GetNamespace(),
-		Name:      cr.Spec.ClusterMasterRef.Name,
-	}
-	masterIdxCluster := &enterprisev1.ClusterMaster{}
-	err = client.Get(context.TODO(), namespacedName, masterIdxCluster)
-	if err == nil {
-		cr.Status.ClusterMasterPhase = masterIdxCluster.Status.Phase
-	} else {
-		cr.Status.ClusterMasterPhase = splcommon.PhaseError
-	}
-
 	// create or update statefulset for the indexers
 	statefulSet, err := getIndexerStatefulSet(client, cr)
 	if err != nil {
 		return result, err
 	}
-	mgr := indexerClusterPodManager{log: scopedLog, cr: cr, secrets: namespaceScopedSecret, newSplunkClient: splclient.NewSplunkClient}
+
```

Comment:
Nit: Does it make sense to move below piece of code to here so we avoid a call to **mgr.Update**?

```
	if mgr.cr.Status.ClusterMasterPhase != splcommon.PhaseReady {
		mgr.log.Error(err, "Cluster Master is not ready yet")
		return splcommon.PhaseError, err
	}
```

Reply:
Can do that but doesn't make much of a difference, anyways you are returning in the beginning.